mergify_engine/queue/merge_train.py

```diff
@@ -407,16 +412,39 @@ async def create_pull(
             ep.user_pull_request_number for ep in self.still_queued_embarked_pulls
         ]:
             try:
-                await self.train.repository.installation.client.post(
-                    f"/repos/{self.train.repository.installation.owner_login}/{self.train.repository.repo['name']}/merges",
-                    json={
-                        "base": branch_name,
-                        "head": f"refs/pull/{pull_number}/head",
-                        "commit_message": f"Merge of #{pull_number}",
-                    },
-                )
+                # NOTE(sileht): From time to time, GitHub returns a 404 when we merge
+                # the pull request in branches because the branch doesn't exists yet
+                # even if the previous API call returns
+                async for attempt in tenacity.AsyncRetrying(
+                    wait=tenacity.wait_exponential(multiplier=0.1),
```

Comment:
Do we need exponential?
Or 4 times every 0.5s is good enough?
(no clue)

Reply:
I checked the log, it only happens 40 times in a week (all org mixed), so I guess a very short retry should be enough to retry fast here ~1.5s and if it still fail, we PostPone the creation of the PR and on next worker pass we will retry.